_pytest/python.py

```diff
@@ -1412,8 +1412,12 @@ def raises(expected_exception, *args, **kwargs):
     elif not isclass(expected_exception):
         raise TypeError(msg % type(expected_exception))
 
+    message = "DID NOT RAISE {0}".format(expected_exception)
+
     if not args:
-        return RaisesContext(expected_exception)
+        if "message" in kwargs:
```

Comment:
Please update the docstring to include this new keyword argument support; make sure to mention it is only valid in the context manager form. :grin: